x-pack/plugins/stack_connectors/public/connector_types/pagerduty/pagerduty.tsx

```diff
@@ -92,18 +93,22 @@ export function getConnectorType(): ConnectorTypeModel<
           }
         });
       }
-      if (actionParams.customDetails?.length && !hasMustacheTokens(actionParams.customDetails)) {
```

Comment:
We still need the `!hasMustacheTokens()` check in here, if we want to support property values that are not strings.  The problem is for such properties, the value shouldn't be quoted.  So, if you wanted a customDetail property of `value` to be the the numeric value `context.value` (for index threshold rule type), you would code this as

```
{ 
  "value": {{context.value}} 
}
```

But with this code, that generates an error:

![image](https://github.com/elastic/kibana/assets/25117/ce0b5944-a521-4f86-a3c0-c65d137b4b08)

We can only validate custom details like that when the executor finally runs.  But of course, that would be a shame to fail the connector execution if that particular field was invalid JSON.   I wonder if would make sense to just not use the custom details if they didn't validate - we'd need to log a message, and see if we could get a message into the UX as well ...

Reply:
The problem with not validating when there are context variables was that invalid json like 
```
{
  {{context.value}}
}
```
was getting passed through as valid as well. This way at least they would get something through even if it is stringified? it's the same problem we have with the ES index connector. I'm not familiar enough with the custom details field to know whether it matters if a numerical value shows up a stringified.